idx/cassandra/cassandra.go

```diff
@@ -371,8 +376,13 @@ func (c *CasIdx) Load(defs []schema.MetricDefinition, cutoff uint32) []schema.Me
 	return c.load(defs, iter, cutoff)
 }
 
-func (c *CasIdx) LoadPartition(partition int32, defs []schema.MetricDefinition, cutoff uint32) []schema.MetricDefinition {
-	iter := c.session.Query("SELECT id, orgid, partition, name, interval, unit, mtype, tags, lastupdate from metric_idx where partition=?", partition).Iter()
+func (c *CasIdx) LoadPartitions(partitions []int32, defs []schema.MetricDefinition, cutoff uint32) []schema.MetricDefinition {
+	placeholders := make([]string, len(partitions))
+	for i, p := range partitions {
+		placeholders[i] = strconv.Itoa(int(p))
+	}
+	q := fmt.Sprintf("SELECT id, orgid, partition, name, interval, unit, mtype, tags, lastupdate from metric_idx where partition in (%s)", strings.Join(placeholders, ","))
+	iter := c.session.Query(q).Iter()
```

Comment:
can you shed some more light on this? is this merely because cosmos doesn't understand the former syntax, or are there other reasons such as performance?

Reply:
This is just an optimization.  I ran into issues on a dev cosmosdb instance due to how it limits requests.  Basically, making the request with a single query rather then separate queries for each partition uses fewer resources.